tools/bootnode/bootnode.go

```diff
@@ -152,8 +155,16 @@ func createListener(ipAddr string, port int, cfg discover.Config) *discover.UDPv
 	if ip.To4() == nil {
 		log.Fatalf("IPV4 address not provided instead %s was provided", ipAddr)
 	}
+	localIP := ip
+	if !ip.IsLoopback() {
+		rawIP, err := iputils.ExternalIPv4()
```

Comment:
I find this a bit confusing, but I understand after discussing offline that this is just used for binding.